winterm/scroll_helper.go

```diff
@@ -4,8 +4,8 @@ package winterm
 
 // effectiveSr gets the current effective scroll region in buffer coordinates
 func (h *windowsAnsiEventHandler) effectiveSr(window SMALL_RECT) scrollRegion {
-	top := addInRange(window.Top, h.sr.top, window.Top, window.Bottom)
-	bottom := addInRange(window.Top, h.sr.bottom, window.Top, window.Bottom)
+	top := ensureInRange(window.Top+h.sr.top, window.Top, window.Bottom)
+	bottom := ensureInRange(window.Top+h.sr.bottom, window.Top, window.Bottom)
```

Comment:
@jstarks  trying to get my head around if this is correct or a bug; wondering if this should be;

```suggestion
	bottom := ensureInRange(window.Bottom+h.sr.bottom, window.Top, window.Bottom)
```

Looks like it was added in https://github.com/Azure/go-ansiterm/commit/396f10f6557209a04f52ef588ffb668042fca010

Reply:
No, I believe it's correct as-is. The effective scroll region is relative to the window rectangle, so adding to the window top is the right thing.

Of course I wrote this code 5 years ago so who knows ;)